app/controllers/api/groups_controller.rb

```diff
@@ -80,5 +80,59 @@ def include_students(fields)
       end
     end
 
+    # Update the group's marks for the given assignment.
+    def update_marks
+      assignment = Assignment.find(params[:assignment_id])
+      if assignment.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+          'No assignment exists with that id' }, status: 404
+        return
+      end
+
+      group = Group.find(params[:id])
+      if group.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+          'No group exists with that id' }, status: 404
+        return
+      end
+      if group.grouping_for_assignment(params[:assignment_id])
+              .has_submission?
+        result = group.grouping_for_assignment(params[:assignment_id])
+                      .current_submission_used
+                      .get_latest_result
+      else
+        render 'shared/http_status', locals: { code: '404', message:
+          'No submissions exist for that group' }, status: 404
+        return
+      end
+
+      matched_criteria = RubricCriterion
+                           .where(rubric_criterion_name: params.keys)
```

Comment:
I think I meant to make a comment here and then didn't. This isn't searching criteria by a particular assignment, which isn't good if two assignments have a criterion with the same name.